--- eeschema/tool_sch.cpp~	2012-01-26 23:08:12.000000000 +0100
+++ eeschema/tool_sch.cpp	2012-01-26 23:08:12.000000000 +0100
@@ -187,15 +187,15 @@
     m_VToolBar->AddTool( ID_NOCONN_BUTT, wxEmptyString, KiBitmap( noconn_xpm ),
                          HELP_PLACE_NC_FLAG, wxITEM_CHECK );
 
+    m_VToolBar->AddTool( ID_JUNCTION_BUTT, wxEmptyString, KiBitmap( add_junction_xpm ),
+                         HELP_PLACE_JUNCTION, wxITEM_CHECK );
+
     m_VToolBar->AddTool( ID_LABEL_BUTT, wxEmptyString, KiBitmap( add_line_label_xpm ),
                          HELP_PLACE_NETLABEL, wxITEM_CHECK );
 
     m_VToolBar->AddTool( ID_GLABEL_BUTT, wxEmptyString, KiBitmap( add_glabel_xpm ),
                          HELP_PLACE_GLOBALLABEL, wxITEM_CHECK );
 
-    m_VToolBar->AddTool( ID_JUNCTION_BUTT, wxEmptyString, KiBitmap( add_junction_xpm ),
-                         HELP_PLACE_JUNCTION, wxITEM_CHECK );
-
     m_VToolBar->AddTool( ID_HIERLABEL_BUTT, wxEmptyString, KiBitmap( add_hierarchical_label_xpm ),
                          HELP_PLACE_HIER_LABEL, wxITEM_CHECK );
 
