--- eeschema/edit_component_in_schematic.cpp.orig	2010-02-18 21:07:29.000000000 +0100
+++ eeschema/edit_component_in_schematic.cpp	2010-04-26 15:49:07.000000000 +0200
@@ -416,13 +416,13 @@
 
         Cmp->GetField( FOOTPRINT )->m_Orient =
             Cmp->GetField( REFERENCE )->m_Orient;
-    }
-    TextField->m_Text = message;
+        TextField->m_Text = message;
 
-    Cmp->GetField( FOOTPRINT )->Draw( DrawPanel, DC, wxPoint( 0, 0 ),
-                                      Cmp->m_Flags ? g_XorMode :
-                                      GR_DEFAULT_DRAWMODE );
-    OnModify( );
+        Cmp->GetField( FOOTPRINT )->Draw( DrawPanel, DC, wxPoint( 0, 0 ),
+                                          Cmp->m_Flags ? g_XorMode :
+                                          GR_DEFAULT_DRAWMODE );
+        OnModify( );
+    }
 
     Cmp->DisplayInfo( this );
 }
