diff -ru kicad-dev/cvpcb/makefile.gtk kicad-dev-fedora/cvpcb/makefile.gtk
--- kicad-dev/cvpcb/makefile.gtk	2006-05-09 08:42:12.000000000 +0200
+++ kicad-dev-fedora/cvpcb/makefile.gtk	2007-02-07 16:39:09.000000000 +0100
@@ -31,9 +31,9 @@
 	$(LD) $(OBJECTS) $(LDFLAGS)  $(LIBVIEWER3D)  $(LIBS_WITH_GL) -o $(TARGET)
 
 
-install:
-	cp -f $(TARGET) $(KICAD_BIN)
-
+install:
+	cp -f $(TARGET) $(DESTDIR)$(KICAD_BIN)
+
 clean:
 		rm *.o
 		rm $(TARGET)
diff -ru kicad-dev/eeschema/makefile.gtk kicad-dev-fedora/eeschema/makefile.gtk
--- kicad-dev/eeschema/makefile.gtk	2006-06-02 13:45:48.000000000 +0200
+++ kicad-dev-fedora/eeschema/makefile.gtk	2007-02-07 16:39:35.000000000 +0100
@@ -31,7 +31,7 @@
 
 
 install:
-	cp $(TARGET) $(KICAD_BIN)
+	cp $(TARGET) $(DESTDIR)$(KICAD_BIN)
 
 
 clean:
diff -ru kicad-dev/gerbview/makefile.gtk kicad-dev-fedora/gerbview/makefile.gtk
--- kicad-dev/gerbview/makefile.gtk	2006-06-02 13:45:54.000000000 +0200
+++ kicad-dev-fedora/gerbview/makefile.gtk	2007-02-07 16:39:47.000000000 +0100
@@ -21,10 +21,10 @@
 
 
 $(TARGET): $(OBJECTS) makefile.gtk makefile.include $(EXTRALIBS) ../libs.linux
-		$(LD) $(LDFLAGS) $(OBJECTS) $(LIBS) -o $(TARGET)
-		
+		$(LD) $(LDFLAGS) $(OBJECTS) $(LIBS) -o $(TARGET)
+		
 install:
-		cp $(TARGET) $(KICAD_BIN)
+		cp $(TARGET) $(DESTDIR)$(KICAD_BIN)
 
 
 clean:
diff -ru kicad-dev/kicad/makefile.gtk kicad-dev-fedora/kicad/makefile.gtk
--- kicad-dev/kicad/makefile.gtk	2006-10-16 13:34:02.000000000 +0200
+++ kicad-dev-fedora/kicad/makefile.gtk	2007-02-07 16:40:08.000000000 +0100
@@ -26,7 +26,7 @@
 	$(LD) $(OBJECTS) $(LDFLAGS)  $(LIBS) -o $(TARGET)
 
 install:
-	cp -f  $(TARGET) $(KICAD_BIN)
+	cp -f  $(TARGET) $(DESTDIR)$(KICAD_BIN)
 
 clean:
 	rm -f *.o;
diff -ru kicad-dev/libs.linux kicad-dev-fedora/libs.linux
--- kicad-dev/libs.linux	2006-11-14 16:01:20.000000000 +0100
+++ kicad-dev-fedora/libs.linux	2007-02-07 17:03:43.000000000 +0100
@@ -1,6 +1,6 @@
 
 #Configuration for build kicad
-KICAD_BIN = /usr/local/kicad/linux
+KICAD_BIN = /usr/bin
 
 
 SRCSUFF = .cpp
@@ -11,10 +11,10 @@
 # You must comment or uncomment this line for dynamic or static link
 # dynamic link is less difficult than static link
 # choose it if you have problems (libs not found, or unresolved references) ), or if kicad is running only on YOUR system.
-KICAD_STATIC_LINK = 1
+# KICAD_STATIC_LINK = 1
 
 # path and version definition for wxWidgets and mesa libs
-MESALIBSPATH = /usr/local/lib
+MESALIBSPATH = /usr/lib
 WXPATH = `wx-config --prefix`/lib
 PREFIX_WX_LIBS = lib`wx-config --basename`
 SUFFIX_WX_LIBSTD = `wx-config --utility=`
@@ -69,11 +69,11 @@
 endif
 
 
-# attention à l'ordre des libairies
-LIBS = -L/usr/local/lib -L/usr/X11R6/lib\
+# attention ï¿½l'ordre des libairies
+LIBS = -L/usr/lib -L/usr/X11R6/lib\
 	$(EXTRALIBS) $(WXSYSLIB)\
 	$(LIBSTDC)
 
-LIBS_WITH_GL = -L/usr/local/lib -L/usr/X11R6/lib\
+LIBS_WITH_GL = -L/usr/lib -L/usr/X11R6/lib\
 	$(EXTRALIBS) $(WXSYSLIB_WITH_GL)\
 	$(LIBSTDC)
Seulement dans kicad-dev-fedora: libs.linux~
diff -ru kicad-dev/pcbnew/makefile.gtk kicad-dev-fedora/pcbnew/makefile.gtk
--- kicad-dev/pcbnew/makefile.gtk	2006-06-02 13:45:50.000000000 +0200
+++ kicad-dev-fedora/pcbnew/makefile.gtk	2007-02-07 16:40:27.000000000 +0100
@@ -23,11 +23,11 @@
 
 $(TARGET): $(OBJECTS) makefile.gtk  makefile.include $(LIBVIEWER3D) $(EXTRALIBS) ../libs.linux
 		$(LD) $(LDFLAGS) $(OBJECTS) $(LIBVIEWER3D) $(LIBS_WITH_GL)\
-		-o $(TARGET)
-		
-		
+		-o $(TARGET)
+		
+		
 install:
-		cp $(TARGET) $(KICAD_BIN)
+		cp $(TARGET) $(DESTDIR)$(KICAD_BIN)
 
 
 clean:
