--- CMakeLists.txt.orig	2010-05-17 23:37:14.000000000 +0200
+++ CMakeLists.txt	2010-05-17 23:42:28.000000000 +0200
@@ -2,9 +2,9 @@
 
 # test the minimum Cmake version requirement (could be different under unix or Windows
 if(WIN32)
-    cmake_minimum_required(VERSION 2.8.0 FATAL_ERROR)
+    cmake_minimum_required(VERSION 2.6.1 FATAL_ERROR)
 else(WIN32)
-    cmake_minimum_required(VERSION 2.8.0 FATAL_ERROR)
+    cmake_minimum_required(VERSION 2.6.0 FATAL_ERROR)
 endif(WIN32)
 
 # Path to local CMake modules.
