--- common/build_version.cpp.orig	2011-03-01 20:24:28.000000000 +0100
+++ common/build_version.cpp	2011-03-02 15:10:11.000000000 +0100
@@ -6,11 +6,11 @@
 #endif
 
 #ifndef KICAD_BUILD_VERSION
-#define KICAD_BUILD_VERSION "(2011-01-26 BZR 2759)"
+#define KICAD_BUILD_VERSION "(2011-01-28 BZR 2765)"
 #endif
 
-//#define VERSION_STABILITY      "stable"
-#define VERSION_STABILITY       "testing"
+#define VERSION_STABILITY      "stable"
+//#define VERSION_STABILITY       "testing"
 
 
 /**
