diff -ru kicad-2010.05.27-6.rev2363/3d-viewer/3d_canvas.cpp kicad-2010.05.27-7.rev2363/3d-viewer/3d_canvas.cpp
--- kicad-2010.05.27-6.rev2363/3d-viewer/3d_canvas.cpp	2010-05-28 21:47:32.000000000 +0200
+++ kicad-2010.05.27-7.rev2363/3d-viewer/3d_canvas.cpp	2011-01-29 15:17:40.000000000 +0100
@@ -45,9 +45,9 @@
                 Pcb3D_GLCanvas::OnPopUpMenu )
 END_EVENT_TABLE()
 
-Pcb3D_GLCanvas::Pcb3D_GLCanvas( WinEDA3D_DrawFrame* parent ) :
-#if wxCHECK_VERSION( 2, 9, 0 )
-    wxGLCanvas( parent, -1, NULL, wxDefaultPosition, wxDefaultSize,
+Pcb3D_GLCanvas::Pcb3D_GLCanvas( WinEDA3D_DrawFrame* parent, int* attribList ) :
+#if wxCHECK_VERSION( 2, 7, 0 )
+    wxGLCanvas( parent, -1, attribList, wxDefaultPosition, wxDefaultSize,
                 wxFULL_REPAINT_ON_RESIZE )
 #else
     wxGLCanvas( parent, -1, wxDefaultPosition, wxDefaultSize,
@@ -58,7 +58,7 @@
     m_gllist = 0;
     m_Parent = parent;
 
-#if wxCHECK_VERSION( 2, 9, 0 )
+#if wxCHECK_VERSION( 2, 7, 0 )
 
     // Explicitly create a new rendering context instance for this canvas.
     m_glRC = new wxGLContext( this );
@@ -72,7 +72,7 @@
 {
     ClearLists();
     m_init = FALSE;
-#if wxCHECK_VERSION( 2, 9, 0 )
+#if wxCHECK_VERSION( 2, 7, 0 )
     delete m_glRC;
 #endif
 }
diff -ru kicad-2010.05.27-6.rev2363/3d-viewer/3d_draw.cpp kicad-2010.05.27-7.rev2363/3d-viewer/3d_draw.cpp
--- kicad-2010.05.27-6.rev2363/3d-viewer/3d_draw.cpp	2010-06-15 22:51:47.000000000 +0200
+++ kicad-2010.05.27-7.rev2363/3d-viewer/3d_draw.cpp	2011-01-29 15:08:42.000000000 +0100
@@ -52,7 +52,11 @@
 
 void Pcb3D_GLCanvas::Redraw( bool finish )
 {
-#if wxCHECK_VERSION( 2, 9, 0 )
+    /* SwapBuffer requires the window to be shown before calling */
+    if( !IsShown() )
+        return;
+
+#if wxCHECK_VERSION( 2, 7, 0 )
     SetCurrent( *m_glRC );
 #else
     SetCurrent();
@@ -93,7 +97,7 @@
     }
 
     glFlush();
-    if( finish )
+    if( finish );
         glFinish();
     SwapBuffers();
 }
diff -ru kicad-2010.05.27-6.rev2363/3d-viewer/3d_frame.cpp kicad-2010.05.27-7.rev2363/3d-viewer/3d_frame.cpp
--- kicad-2010.05.27-6.rev2363/3d-viewer/3d_frame.cpp	2010-05-28 21:47:32.000000000 +0200
+++ kicad-2010.05.27-7.rev2363/3d-viewer/3d_frame.cpp	2011-01-29 15:10:05.000000000 +0100
@@ -78,7 +78,8 @@
     ReCreateVToolbar();
 
     // Make a Pcb3D_GLCanvas
-    m_Canvas = new Pcb3D_GLCanvas( this );
+    int attrs[] = { WX_GL_RGBA, WX_GL_DOUBLEBUFFER, WX_GL_DEPTH_SIZE, 16, 0 };
+    m_Canvas = new Pcb3D_GLCanvas( this, attrs );
 
     m_auimgr.SetManagedWindow( this );
 
diff -ru kicad-2010.05.27-6.rev2363/3d-viewer/3d_viewer.h kicad-2010.05.27-7.rev2363/3d-viewer/3d_viewer.h
--- kicad-2010.05.27-6.rev2363/3d-viewer/3d_viewer.h	2010-05-28 21:47:32.000000000 +0200
+++ kicad-2010.05.27-7.rev2363/3d-viewer/3d_viewer.h	2011-01-29 15:10:51.000000000 +0100
@@ -137,11 +137,11 @@
 private:
     bool         m_init;
     GLuint       m_gllist;
-#if wxCHECK_VERSION( 2, 9, 0 )
+#if wxCHECK_VERSION( 2, 7, 0 )
     wxGLContext* m_glRC;
 #endif
 public:
-    Pcb3D_GLCanvas( WinEDA3D_DrawFrame* parent );
+    Pcb3D_GLCanvas( WinEDA3D_DrawFrame* parent, int* attribList = 0 );
     ~Pcb3D_GLCanvas();
 
     void   ClearLists();
