--- pcbnew/plotps.cpp.orig	2012-01-22 17:23:38.000000000 +0100
+++ pcbnew/plotps.cpp	2012-01-23 22:00:58.000000000 +0100
@@ -100,7 +100,7 @@
     PS_PLOTTER* plotter = new PS_PLOTTER();
     plotter->set_paper_size( SheetPS );
     plotter->set_scale_adjust( g_PcbPlotOptions.m_FineScaleAdjustX,
-                               g_PcbPlotOptions.m_FineScaleAdjustX );
+                               g_PcbPlotOptions.m_FineScaleAdjustY );
     plotter->set_viewport( offset, scale, g_PcbPlotOptions.m_PlotMirror );
     plotter->set_default_line_width( g_PcbPlotOptions.m_PlotLineWidth );
     plotter->set_creator( wxT( "PCBNEW-PS" ) );
