--- pcbnew/dialog_design_rules.cpp.orig	2010-03-02 12:54:11.000000000 +0100
+++ pcbnew/dialog_design_rules.cpp	2010-04-29 20:03:28.000000000 +0200
@@ -294,8 +294,12 @@
     if( a->clazz < b->clazz )
         return true;
 
-    if( a->net < b->net )
-        return true;
+    // inside the same class, sort by net name:
+    if( a->clazz == b->clazz )
+    {
+        if( a->net < b->net )
+            return true;
+    }
 
     return false;
 }
