diff -r -u old/scripts/bom-in-python/ky/ky.py new/scripts/bom-in-python/ky/ky.py
--- old/scripts/bom-in-python/ky/ky.py	2012-01-21 19:01:36.000000000 +0100
+++ new/scripts/bom-in-python/ky/ky.py	2012-01-21 21:43:23.464126596 +0100
@@ -298,11 +298,11 @@
             self.libparts.append(part(self._curr_element))
 
         # If this element is a net, add it to the nets list
-        if self._curr_element.name = "net"
+        if self._curr_element.name == "net":
             self.nets.append(self._curr_element)
         
         # If this element is a library, add it to the libraries list
-        if self._curr_element.name = "library"
+        if self._curr_element.name == "library":
             self.libraries.append(self._curr_element)
                 
         return self._curr_element        
