--- pcb_calculator/CMakeLists.txt.orig	2012-01-22 17:23:40.000000000 +0100
+++ pcb_calculator/CMakeLists.txt	2012-01-23 20:52:58.000000000 +0100
@@ -62,7 +62,7 @@
         ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist)
 endif(APPLE)
 
-target_link_libraries( pcb_calculator common polygon bitmaps
+target_link_libraries( pcb_calculator common polygon kbool bitmaps
                         ${wxWidgets_LIBRARIES}
                         )
 
