diff -Naur kicad-2014.03.13.orig/resources/linux/mime/applications/pcbcalculator.desktop kicad-2014.03.13/resources/linux/mime/applications/pcbcalculator.desktop
--- kicad-2014.03.13.orig/resources/linux/mime/applications/pcbcalculator.desktop	2014-03-13 16:43:01.000000000 +0100
+++ kicad-2014.03.13/resources/linux/mime/applications/pcbcalculator.desktop	2014-03-17 15:40:44.201939540 +0100
@@ -2,7 +2,7 @@
 [Desktop Entry]
 Categories=Development;Electronics
 Comment=Design a printed circuit board
-Exec=pcbcalculator
+Exec=pcb_calculator
 GenericName=EDA Suite
 Icon=pcbcalculator
 MimeType=application/x-pcbcalculator-project;
